### PR TITLE
Investigate available delivery location types method

### PR DESCRIPTION
**Pull request by ktp242**
This PR focus on four aspects,

1. It adds the check for catch the error when PType is not on the list. Now it will correctly assign the location as "Research". See "lib/available_delivery_location_types.js" line 62 -115
2. It also adds logging if it encounters the unrecognizable PType. See See "lib/available_delivery_location_types.js" line 85
3. Extending from point 2, it adds the mechanism to collect all the unfamiliar PType and make a list of it. It will log the list with the unfamiliar PType. Also, when a PType is recognizable, it will then remove it from the list. See "lib/available_delivery_location_types.js". Not sure if this is over kill, but I am happy to discuss :-)
4. It deletes the config and documentations about development tier

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -17,4 +17,4 @@ NYPL_OAUTH_ID=who-you-will-connect-to-the-api-as
 NYPL_OAUTH_SECRET=that-accounts-pw
 NYPL_CORE_VERSION=v1.21
 
-CATALOG_WEBPUB_DEF_URL=[fqdn to catalog webpub.def]
+CATALOG_WEBPUB_DEF_URL=[fqdn to catalog webpub.def. Only required for running scripts/update-annotated-marc-rules.js]
```
Comment (ktp242) [inline .env.example]: Not sure if this is the good explanations. Suggestions are welcome!

File: lib/available_delivery_location_types.js [hunk 2]
```diff
@@ -34,8 +59,63 @@ class AvailableDeliveryLocationTypes {
...
+      })
+      .catch((e) => {
+        let errorMessage = `Error checking unfamiliar Patron Type list: ${e}`
+        logger.error(errorMessage)
+        throw new Error(errorMessage)
+      })
+  }
 }
 
+// Why line 42 repeats line 10
```
Comment (ktp242) [inline lib/available_delivery_location_types.js]: I am not sure why 119 and 121 repeat the code we have before. This just serves as a mark I will remove it before merging.

File: lib/delivery-locations-resolver.js [hunk 3]
```diff
@@ -93,7 +93,7 @@ class DeliveryLocationsResolver {
 
   static resolveDeliveryLocations (items, deliveryLocationTypes) {
     // Assert sensible default for location types:
-    if (!deliveryLocationTypes || !Array.isArray(deliveryLocationTypes)) deliveryLocationTypes = ['Research']
+    if (!Array.isArray(deliveryLocationTypes) || deliveryLocationTypes.length === 0) deliveryLocationTypes = ['Research']
```
Comment (ktp242) [inline lib/delivery-locations-resolver.js]: I still keep this check because I think it is actually more comprehensive.

**Comment by ktp242** [comment]
I noticed there is an error here because one of the function has a too many lines. Based on the development policies of this application, should I modify it or it is more just a warning?

**Comment by nonword** [comment]
I've removed the codeclimate webhook. It's a nice service, but I don't think it makes sense to run our code through climate on-push until it's a part of our local testing practice; Learning of a style issue *after* one pushes is the wrong time to learn. If/when we add codeclimate as part of local testing, we can reenable the hook. Ignore the error.

**Comment by ktp242** [comment]
Thanks Paul!

**Comment by ktp242** [comment]
I have updated the codes to remove the related functions and json about creating the list of unfamiliar Patron Types.

**Comment by ktp242** [comment]
Thanks Paul!

**Pull request merged**